backend/src/BuiltinExecution/Libs/List.fs

```diff
@@ -74,6 +75,13 @@ module DvalComparator =
     | DEnum _, _ ->
       // TODO: Feels like this should hook into typechecker and ValueTypes somehow
       raiseString "Comparing different types" [ "dv1", dv1; "dv2", dv2 ]
+  and compareLetPatternsLists
+    (l1 : NEList<LetPattern>)
+    (l2 : NEList<LetPattern>)
+    : int =
+    let l1 = NEList.map RT2DT.LetPattern.toDT l1 |> NEList.toList
```

Comment:
I don't think this is a good approach, partially because we don't really want to introduce RT2DT in here, but also because it has irrelevant details like the `id` field, which isn't important (and will never be equal). So I would use a `match` and establish an ordering for eg vars vs tuples.